staging/src/k8s.io/kubectl/pkg/cmd/debug/profiles.go

```diff
@@ -49,11 +61,154 @@ func profileLegacy(pod *corev1.Pod, containerName string, target runtime.Object)
 			})
 		}
 
-		pod.Spec.HostIPC = true
 		pod.Spec.HostNetwork = true
 		pod.Spec.HostPID = true
+		pod.Spec.HostIPC = true
 		return nil
 	default:
 		return fmt.Errorf("the %s profile doesn't support objects of type %T", ProfileLegacy, target)
 	}
 }
+
+func (p *generalProfile) Apply(pod *corev1.Pod, containerName string, target runtime.Object) error {
+	switch t := target.(type) {
+	case *corev1.Pod:
```

Comment:
There's going to be a lot of duplication in the profiles, so I recommend creating some helper functions to make this easier to read.

I had imagined something like this:

```
switch t:= target.(type) {
case *corev1.Pod:
  setPtrace(pod, containerName)
  stripAllProbes(pod)
case *corev1.Node:
  SetHostNamespaces(pod)
}
```

What do you think?

Reply:
I think that makes sense - what do you think of c10aec8b092ad1428deb05f76697466348d27e20